instrumentation/restlet/restlet-1.0/library/src/main/java/io/opentelemetry/instrumentation/restlet/v1_0/RestletSingletons.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v1_0;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanStatusExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpSpanStatusExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetAttributesExtractor;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+
+public final class RestletSingletons {
```

Comment:
For library instrumentations we don't really have `*Singleton` classes - instead we create `*Tracing`/`*TracingBuilder` classes that serve as the public API of the instrumentation and allow it to be customized (e.g. passing `OpenTelemetry`, custom `AttributesExtractor`s).
Please take a look at `OkHttpTracing` and its builder for an example of this.

Reply:
So this class should be moved to the `javaagent` module and `library` should have `*Tracing` and `*TracingBuilder`?

Reply:
Yes please 👍